prov/rxm/src/rxm.h

```diff
@@ -421,10 +421,50 @@ struct rxm_conn {
 	struct dlist_entry posted_rx_list;
 	struct dlist_entry sar_rx_msg_list;
 	struct util_cmap_handle handle;
-	/* This is saved MSG EP fid, that hasn't been closed during
-	 * handling of CONN_RECV in CMAP_CONNREQ_SENT for passive side */
-	struct fid_ep *saved_msg_ep;
-	struct dlist_entry saved_posted_rx_list;
+
+	struct rxm_conn_close *close_data;
+};
+
+struct rxm_conn_close {
+	struct fid_ep *msg_ep;
+	struct dlist_entry posted_rx_list;
+};
+
+enum rxm_cmd_internal_type {
+	RXM_CMD_CONNREQ_ACCEPT,
+	RXM_CMD_CONNECTED,
+	RXM_CMD_CONN_CLOSE,
+	RXM_CMD_CONN_FREE,
+};
+
+struct rxm_cmd_connreq_data {
+	/* Must stay at top */
+	enum rxm_cmd_internal_type type; 
+
+	struct rxm_conn *rxm_conn;
+	struct fi_info *msg_info;
+	struct rxm_cm_data cm_data;
+};
+
+struct rxm_cmd_connect_data {
+	/* Must stay at top */
+	enum rxm_cmd_internal_type type; 
+
+	struct rxm_conn *rxm_conn;
+};
+
+struct rxm_cmd_conn_close_data {
+	/* Must stay at top */
+	enum rxm_cmd_internal_type type; 
+
+	struct rxm_conn_close *close_data;
+};
+
+struct rxm_cmd_conn_free_data {
+	/* Must stay at top */
+	enum rxm_cmd_internal_type type; 
+
+	struct rxm_conn *rxm_conn;
 };
```

Comment:
Can we have a single common data struct? We can use an union inside to store different types of data.

Reply:
Sure, nice point.